lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/XmlReader.java

```diff
@@ -0,0 +1,89 @@
+package io.openliberty.tools.langserver.lemminx.services;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.logging.Logger;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+public class XmlReader {
+    private static final Logger LOGGER = Logger.getLogger(XmlReader.class.getName());
+    private static XmlReader instance;
+
+    public static XmlReader getInstance() {
+        if (instance == null) {
+            instance = new XmlReader();
+        }
+        return instance;
+    }
+
+    public boolean hasServerRoot(File file) {
+        if (!file.exists() || file.length() == 0) {
+            return false;
+        }
+
+        XMLInputFactory factory = XMLInputFactory.newInstance();
+        try {
+            XMLEventReader reader = factory.createXMLEventReader(new FileInputStream(file));
+            if (reader.hasNext()) {
+                XMLEvent firstTag = reader.nextTag(); // first start/end element
+                reader.close();
+                return isServerElement(firstTag);
+            }
+            reader.close();
```

Comment:
maybe define `reader` outside of the try/catch and then close it in a `finally` if it is not null